packages/neuron-ui/src/components/History/index.tsx

```diff
@@ -36,19 +47,38 @@ const transactionsToHistory = (transactions: Transaction[]) =>
   }))
 
 const History = () => {
-  const chain = useContext(ChainContext)
   const pageSize = 14
+  const initTransactions: TxHistory[] = []
+
+  const chain = useContext(ChainContext)
   const [page, setPage] = useState(0)
+  const [items, setItems] = useState(initTransactions)
+  const [loading, setLoading] = useState(false)
 
   useEffect(() => {
     // This should be moved to the top level
     getTransactions(page, pageSize)
-  }, [page, pageSize])
+  }, [page])
+
+  useEffect(() => {
+    const newItems = transactionsToHistory(chain.transactions.items)
+    setItems(page === 0 ? newItems : items.concat(newItems))
+    setLoading(false)
+  }, [chain])
+
+  const onMore = () => {
+    const element = document.getElementById('root') as HTMLElement
+    const scrollBottom = element.scrollHeight - element.offsetHeight - element.scrollTop
```

Comment:
Doesn't look a good idea to manipulate DOM. We should avoid this as much as possible.